scipy/signal/_signaltools.py

```diff
@@ -3540,170 +3566,152 @@ def resample(x, num, t=None, axis=0, window=None, domain='time'):
+    intervals. For non-periodic signals, `resample_poly` may be a better choice.
+    Consult the `scipy.interpolate` module for methods of resampling signals with
+    non-constant sampling intervals.
 
     Examples
     --------
-    Note that the end of the resampled data rises to meet the first
-    sample of the next cycle:
+    The following example depicts a signal being up-sampled from 20 samples to 100
+    samples. The ringing at the beginning of the up-sampled signal is due to
+    interpreting the signal being periodic. The red square in the plot illustrates that
+    periodictiy by showing the first sample of the next cycle of the signal.
 
     >>> import numpy as np
-    >>> from scipy import signal
-
-    >>> x = np.linspace(0, 10, 20, endpoint=False)
-    >>> y = np.cos(-x**2/6.0)
-    >>> f = signal.resample(y, 100)
-    >>> xnew = np.linspace(0, 10, 100, endpoint=False)
-
     >>> import matplotlib.pyplot as plt
-    >>> plt.plot(x, y, 'go-', xnew, f, '.-', 10, y[0], 'ro')
-    >>> plt.legend(['data', 'resampled'], loc='best')
+    >>> from scipy.signal import resample
+    ...
+    >>> n0, n1 = 20, 100  # number of samples
+    >>> t0 = np.linspace(0, 10, n0, endpoint=False)  # input time stamps
+    >>> x0 = np.cos(-t0**2/6)  # input signal
+    ...
+    >>> x1 = resample(x0, n1)  # resampled signal
+    >>> t1 = np.linspace(0, 10, n1, endpoint=False)  # timestamps of x1
+    ...
+    >>> fig0, ax0 = plt.subplots(1, 1, tight_layout=True)
+    >>> ax0.set_title(f"Resampling $x(t)$ from {n0} samples to {n1} samples")
+    >>> ax0.set(xlabel="Time $t$", ylabel="Amplitude $x(t)$")
+    >>> ax0.plot(t1, x1, '.-', alpha=.5, label=f"Resampled")
+    >>> ax0.plot(t0, x0, 'o-', alpha=.5, label="Original")
+    >>> ax0.plot(10, x0[0], 'rs', alpha=.5, label="Next Cycle")
+    >>> ax0.legend(loc='best')
+    >>> ax0.grid(True)
     >>> plt.show()
 
-    Consider the following signal  ``y`` where ``len(y)`` is a large  prime number:
-
-    >>> N = 55949
-    >>> freq = 100
-    >>> x = np.linspace(0, 1, N)
-    >>> y = np.cos(2 * np.pi * freq * x)
 
-    Due to ``N`` being prime,
+    The following example illustrates that `~scipy.fft.rfft` / `~scipy.fft.irfft`
+    combination does not always produce the correct resampling result:
 
-    >>> num = 5000
-    >>> f = signal.resample(signal.resample_poly(y, 1, N // num), num)
+    >>> from scipy.fft import irfft, rfft
+    >>> from scipy.signal import resample
+    ...
+    >>> n0, n1 = 8, 4  # number of samples
+    >>> x0 = irfft([0, 0, 8, 0, 0], n=n0)  # input signal
+    >>> x1 = resample(x0, num=n1)
+    >>> y1 = irfft(rfft(x0), n=n1) * n1/n0
+    >>> for x_, n_ in zip((x0, x1, y1), ('x0', 'x1', 'y1')):
+    ...    print(f"{n_} = {x_}")
+    x0 = [ 2.  0. -2.  0.  2.  0. -2.  0.]
+    x1 = [ 2. -2.  2. -2.]
+    y1 = [ 1. -1.  1. -1.]
+
+    It can be easily verified that ``x1`` is correct and ``y1`` is not. To produce
```

Comment:
Consider showing how to verify it.

If the docstring gets too long, consider moving some material into the tutorial, and linking from here.

Reply:
Reworked example in commit 76636f9e7.